InnerEye/ML/run_ml.py

```diff
@@ -752,37 +750,25 @@ def copy_file(source: Path, destination_file: str) -> None:
     def model_inference_train_and_test(self,
                                        checkpoint_handler: CheckpointHandler,
                                        model_proc: ModelProcessing = ModelProcessing.DEFAULT) -> \
-            Tuple[Optional[InferenceMetrics], Optional[InferenceMetrics], Optional[InferenceMetrics]]:
-        train_metrics = None
-        val_metrics = None
-        test_metrics = None
+            Dict[ModelExecutionMode, InferenceMetrics]:
+        metrics: Dict[ModelExecutionMode, InferenceMetrics] = {}
 
         config = self.innereye_config
 
-        def run_model_test(data_split: ModelExecutionMode) -> Optional[InferenceMetrics]:
-            return model_test(config, data_split=data_split, checkpoint_handler=checkpoint_handler,  # type: ignore
-                              model_proc=model_proc)
-
-        if config.perform_validation_and_test_set_inference:
-            # perform inference on test set
-            test_metrics = run_model_test(ModelExecutionMode.TEST)
-            # perform inference on validation set (not for ensemble as current val is in the training fold
-            # for at least one of the models).
-            if model_proc != ModelProcessing.ENSEMBLE_CREATION:
-                val_metrics = run_model_test(ModelExecutionMode.VAL)
-
-        if config.perform_training_set_inference:
-            # perform inference on training set if required
-            train_metrics = run_model_test(ModelExecutionMode.TRAIN)
+        for data_split in ModelExecutionMode:
+            if self.container.run_perform_inference(model_proc, data_split):
```

Comment:
each time I read "run_perform_inference", I get slightly confused. Rename maybe?

Reply:
Changed to "inference_on_set"